Minio/DataModel/Notification/BucketNotification.cs

```diff
@@ -0,0 +1,156 @@
+        public void AddQueue(QueueConfig queueConfig)
+        {
+            for (int index = 0; index < this.QueueConfigs.Count; index++)
+            {
+                if (this.QueueConfigs[index].Equals(queueConfig))
+                    return;
+            }
+            this.QueueConfigs.Add(queueConfig);
+        }
+
+        // AddLambda adds a given lambda config to the general bucket notification config
+        public void AddLambda(LambdaConfig lambdaConfig)
+        {
+            for (int index = 0; index < this.LambdaConfigs.Count; index++)
+            {
+                if (this.LambdaConfigs[index].Equals(lambdaConfig))
+                    return;
+            }
+            this.LambdaConfigs.Add(lambdaConfig);
+        }
+
+        // RemoveTopicByArn removes all topic configurations that match the exact specified ARN
+        public void RemoveTopicByArn(Arn topicArn)
+        {
+            for (int index = 0; index < this.TopicConfigs.Count; index++)
+            {
+                if (this.TopicConfigs[index].Topic.Equals(topicArn.ToString()))
+                {
+                    this.TopicConfigs.RemoveAt(index);
+                    return;
+                }
+            }
+        }
+
+        // RemoveQueueByArn removes all queue configurations that match the exact specified ARN
+        public void RemoveQueueByArn(Arn queueArn)
+        {
+            for (int index = 0; index < this.QueueConfigs.Count; index++)
+            {
+                if (this.QueueConfigs[index].Queue.Equals(queueArn.ToString()))
+                {
+                    this.QueueConfigs.RemoveAt(index);
+                    return;
+                }
+            }
+        }
+
+        // RemoveLambdaByArn removes all lambda configurations that match the exact specified ARN
+        public void RemoveLambdaByArn(Arn lambdaArn)
+        {
+            for (int index = 0; index < this.LambdaConfigs.Count; index++)
+            {
+                if (this.LambdaConfigs[index].Lambda.Equals(lambdaArn.ToString()))
+                {
+                    this.LambdaConfigs.RemoveAt(index);
+                    return;
+                }
+            }
+        }
+        // Helper methods to guide XMLSerializer
+        public bool ShouldSerializeLambdaConfigs()
+        {
+            return LambdaConfigs.Count > 0;
+        }
+        public bool ShouldSerializeTopicConfigs()
+        {
+            return TopicConfigs.Count > 0;
+        }
+        public bool ShouldSerializeQueueConfigs()
+        {
+            return QueueConfigs.Count > 0;
+        }
+
+        public bool ShouldSerializeName()
+        {
+            return this.Name != null;
+        }
+        // Serializes the notification configuration as an XML string
+        public string ToXML()
+        {
+            XmlWriterSettings settings = new XmlWriterSettings();
+            settings.OmitXmlDeclaration = true;
+            MemoryStream ms = new MemoryStream();
```

Comment:
Does it make sense to use `using` block to make `MemoryStream` finalization automagic even in the event of exceptions?